[dogstatsd] document using multiple pipelines to limit packets drop (#13837)
DataDog/documentation: content/en/developers/dogstatsd/high_throughput.md

{{< /programming-lang-wrapper >}}

### Run multiple metrics processing pipeline to limit packet drops

> If you observe that your DogStatsD server using UDS is dropping packets at a high throughput, configuring it to use more CPU could help processing packets faster, and more of them.

Author: Hey @alai97, thanks a lot for the review 🙇 

What do you think of:

```
If your DogStatsD server uses UDS and is dropping packets at a high throughput, configuring the server to use more CPU may improve processing speed and decrease packet drops.
```

> Symptoms are the client-side telemetry indicating that the client is dropping packets in its queue and the DogStatsD server not using more than 2 CPUs/2 cores while the server has more available.

Reviewer: ```suggestion
You may also configure your DogStatsD server if the server does not use more than 2 CPUs or 2 cores even if they are available.
```

Author: We're losing the client-side telemetry information here which is important.

What do you think of:

```
You may also configure your DogStatsD server if the client telemetry indicates packet drops and the server does not use more than 2 CPUs or 2 cores even if they are available.
```

> In order to reduce the amount of drops:

Reviewer: ```suggestion
To reduce the amount of packet drops:

```

> - increase the client queue size to `8192`, please refer to the client library configuration to proceed. Once done, you should already observe the drops amount decrease, but your application may use slightly more RAM.
> - if first step is not enough, you can enable the feature `dogstatsd_pipeline_autoadjust: true` in your Datadog Agent configuration: if available, multiple cores will be used to process the custom metrics, leading to a higher CPU usage but the packet drops should be lower.

Reviewer: Is there a link to send users to the client library configuration?

```suggestion
1. Increase the client queue size to `8192`. For more information, see the client library configuration. You may see the amount of drops decrease, and your application may use more RAM.
2. Additionally, you can enable the `dogstatsd_pipeline_autoadjust: true` feature in your Datadog Agent configuration. The Agent uses multiple cores to process custom metrics, which may lead to higher CPU usage but lowers packet drops.
```
